src/rpc/rawtransaction.cpp

```diff
@@ -820,9 +825,13 @@ UniValue signrawtransaction(const JSONRPCRequest& request)
     // Sign what we can:
     for (unsigned int i = 0; i < mergedTx.vin.size(); i++) {
         CTxIn& txin = mergedTx.vin[i];
+        CScriptWitness witness;
+        if (txin.scriptWitness.stack.size() >= i + 1) {
```

Comment:
I think this is the wrong check. You're testing the size of the witness stack for this particular CTxIn, but I think what you intend is to check for the existence of a witness for this CTxIn. I think the test you need is:

`if (!txin.scriptWitness.IsNull()) {`

Your test will fail if (for example) TxIn in postion 3 has only 2 items on its witness stack.